### PR TITLE
Fix friendly fire striderdetriment.lua

### PR DESCRIPTION
**Pull request by SteelBlueZK**
added variables firetracker and gausslefttrue

the AA gun still switches wrong but I have no idea how to fix it(it fires too fast for endburst) and I do not belive it matters. At least it still switches like before.

### CODE REVIEW

File: scripts/striderdetriment.lua [hunk 1]
```diff
@@ -221,23 +225,49 @@ function script.AimWeapon(num, heading, pitch)
...
+	if num == 2 then
+		gunIndex[num] = (gunIndex[num] % #gunFlares[num]) + 1
+	end
+	flareIndex[num] = gunIndex[num] -- store current gunIndex in flareIndex
+
+end
+
+function script.EndBurst(num)
+	if num == 1 then
+		gunIndex[num] = (gunIndex[num] % #gunFlares[num]) +1
```
Comment (SteelBlueZK) [inline scripts/striderdetriment.lua]: #gunFlares[num] always results in 6 when num == 1

File: scripts/striderdetriment.lua [hunk 2]
```diff
@@ -221,23 +225,49 @@ function script.AimWeapon(num, heading, pitch)
...
 	if num ~= 2 then
 		lastTorsoHeading = heading
 	end
-	
+
 	return true
 end
 
+
+function script.FireWeapon(num)
```
Comment (SteelBlueZK) [inline scripts/striderdetriment.lua]: this whole function is redundant unless somehow Shot is not called 3 times exacly for the gauss burst. and lines 241-246 do nothing.

File: scripts/striderdetriment.lua [hunk 3]
```diff
@@ -221,23 +225,49 @@ function script.AimWeapon(num, heading, pitch)
...
+			gunIndex[1] = firetracker + 1
+		else
+			gunIndex[1] = firetracker + 4
+		end
+		Move(barrels[gunIndex[1] ], z_axis, -20)
+		Move(barrels[gunIndex[1] ], z_axis, 0, 20)
+		firetracker = (firetracker +1) % 3
+	end
+	if num == 2 then
+		gunIndex[num] = (gunIndex[num] % #gunFlares[num]) + 1
```
Comment (SteelBlueZK) [inline scripts/striderdetriment.lua]: #gunFlares[num] always equals 2 when num == 2

File: scripts/striderdetriment.lua [hunk 4]
```diff
@@ -78,12 +78,16 @@ local reloadTime = wd and WeaponDefs[wd].reload*30 or 30
 wd = UnitDefs[unitDefID].weapons[1] and UnitDefs[unitDefID].weapons[1].weaponDef
 local reloadTimeShort = wd and WeaponDefs[wd].reload*30 or 30
 
+
```
Comment (SteelBlueZK) [inline scripts/striderdetriment.lua]: 2 extra variable here

**Comment by GoogleFrog** [comment]
Fixed https://github.com/ZeroK-RTS/Zero-K/commit/79820ee93455863ede643637aeeaaf62ef13c3a6

**Pull request closed without merging**